service.ts

```diff
@@ -0,0 +1,69 @@
+import express from 'express';
+import crypto from 'node:crypto';
+import bodyParser from 'body-parser';
+import { createValidator } from 'express-joi-validation';
+import {
+  createUserBodySchema,
+  paramsIDSchema,
+  paramsSubstringLimitSchema,
+  updateUserBodySchema,
+} from './schemas';
+import type { User } from './types';
+
+const app = express();
+const validator = createValidator();
+
+app.listen(3000);
+
+const users: User[] = [];
+
+app.use(bodyParser.json());
+
+// get all users
+app.get('/users', (req, res) => {
+  res.json(users);
+});
+
+// get user by ID
+app.get('/user/:id', validator.params(paramsIDSchema), (req, res) => {
+  res.json(users.filter((user) => user.id === req.params.id));
```

Comment:
Actually, to res.json() we usually pass variable, not the whole service function. You'll read about layered architecture for the next task, yet for this one you still better use separate variable to get data, and then pass this variable to res.

Reply:
fixed